app/public/js/home.js

```diff
@@ -60,11 +60,53 @@ function initMap()
 		google.maps.event.trigger(map, "resize");
 		map.setCenter(center); 
 	});
+	google.maps.event.addListener(map, 'click', function(event) {
+    locationInfo(map, event.latLng);
+  	});
 
 	conditionalProcessAddress();
 
 	$('#address').bind('keyup change', delayedProcessAddress);	
 }
+//To calculate the radian value of a point
+function rad(pt) {
+  return pt * Math.PI / 180;
+}
+// To calculate the distance between two points on a map 
+function calculateDistance(pt1,pt2){
+  var R = 6378137; // Earth’s mean radius in meter
+  var dLat = rad(pt2.lat() - pt1.lat());
+  var dLong = rad(pt2.lng() - pt1.lng());
+  var a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
+    Math.cos(rad(pt1.lat())) * Math.cos(rad(pt2.lat())) *
+    Math.sin(dLong / 2) * Math.sin(dLong / 2);
+  var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+  var d = R * c;
+  return d; // returns the distance in meter
+}
+/*
+Maps the clicked coordinates on a map to an addresse and set that addresse as a current location
+*/
+function locationInfo(map, location) {
+	var geocoder = new google.maps.Geocoder();
+        var latlng = {lat: location.lat(), lng: location.lng()};
+        geocoder.geocode({'location': latlng}, function(results, status) {
+          if (status === 'OK') {
+            if (results[0]) {
+        		if (calculateDistance(location,results[0].geometry.location)<=100){
+        			document.getElementById("address").value=results[0].formatted_address;
+        			processAddress();
+        		} else {
+        			document.getElementById("address").value="("+location.lat().toFixed(2)+","+location.lng().toFixed(2)+")";
+        		}
+            } else {
+            	throw new Error('No Results');
```

Comment:
Uncaught exceptions in JavaScript can cause unrelated JavaScript logic to break so I want to avoid throwing exceptions when there's another reasonable way to handle the weird cases.

Here, I'd rather just log the error and do nothing than throw.  This way, if we run into the problem, we can see it as developers in the JavaScript console but it won't hurt the end user's experience as badly.

Convert these throw statements to console.error such as: 
console.error('No Results');

and:
console.error('Geocode failed for the following reason: '+status);

Reply:
->i will check the indentation again as it seems good in my text editor but not here.
->and will change these exceptions to console.error as well.